app/models/piece.rb

```diff
@@ -1,7 +1,7 @@
-require 'pry'
 class Piece < ActiveRecord::Base
   belongs_to :user
   belongs_to :game
+  has_many :types
```

Comment:
Other than my quibble with `has_many :types` in piece.rb, this looks solid, Chris. Thanks for putting in those validations, cleaning up the rook tests, and such. Giving it a :+1: